beacon-light-client/solidity/contracts/validators_accumulator/ValidatorsAccumulator.sol

```diff
@@ -125,32 +123,34 @@ contract ValidatorsAccumulator is IValidatorsAccumulator {
   }
 
   function _binarySearchBlock(
-    uint256 blockNumber
-  ) internal view returns (uint256) {
-    uint256 lower = startIndex;
-    uint256 upper = validatorsCount - 1;
-
-    if (snapshots[upper].blockNumber <= blockNumber) {
-      return upper;
+    uint64 blockNumber
+  ) internal view returns (uint64) {
+    uint256 lower;
+    uint256 upper = blockNumbers.length - 1;
+
+    uint64 upperBlockNumber = blockNumbers[upper];
+    if (upperBlockNumber <= blockNumber) {
+      return upperBlockNumber;
     }
 
-    if (snapshots[lower].blockNumber > blockNumber) {
-      return 0;
+    uint64 lowerBlockNumber = blockNumbers[lower];
+    if (lowerBlockNumber > blockNumber) {
+      return lowerBlockNumber;
```

Comment:
In my opinion, it is better to return type(uint64).max as this is a special case where no block was found. And will make the code a bit easier to follow